### PR TITLE
Add Backbone.js

### PR DESCRIPTION
**Pull request by blizzz**
Solves #9227 ← don't merge until we have an agreement there

### CODE REVIEW

File: apps/files/index.php [hunk 1]
```diff
@@ -51,6 +51,7 @@
 \OCP\Util::addScript('files', 'favoritesplugin');
 
 \OC_Util::addVendorScript('core', 'handlebars/handlebars');
+\OC_Util::addVendorScript('core', 'backbone/backbone-min');
```
Comment (PVince81) [inline apps/files/index.php]: We should include the non-minified version instead. It makes it easier to debug.
Also OC has its own minifying code (assets pipeline).

**Comment by PVince81** [comment]
Please add the backbone-min.js and .bower.json files to core/vendor/.gitignore (this was done for other libraries too).

Also I wonder if we should namespace it to OC.Backbone by calling `OC.Backbone = Backbone.noConflict();` to avoid conflict with Backbone libs brought by apps in case the version is different.

**Comment by blizzz** [comment]
> Also I wonder if we should namespace it to OC.Backbone by calling OC.Backbone = Backbone.noConflict(); to avoid conflict with Backbone libs brought by apps in case the version is different.

I have not seen an issue with Mails, but still that's probably better

**Comment by blizzz** [comment]
Being loaded in core now.

**Comment by blizzz** [comment]
Jenkins branch goes here https://github.com/owncloud/core/pull/18097

**Comment by PVince81** [comment]
Looks good :+1: 

Please review @DeepDiver1975 @karlitschek @MorrisJobke @LukasReschke @rullzer @schiesbn

**Comment by PVince81** [comment]
I feel like after this programming in JS will finally be fun again (and also consistent, etc)

**Comment by DeepDiver1975** [comment]
> Please review @DeepDiver1975 @karlitschek @MorrisJobke @LukasReschke @rullzer @schiesbn

I'm all in for backbone :+1:

**Comment by rullzer** [comment]
Looking good! :+1:

**Pull request merged**